jetty-client/src/main/java/org/eclipse/jetty/client/util/DigestAuthentication.java

```diff
@@ -211,8 +211,8 @@ public void apply(Request request)
             String A1 = user + ":" + realm + ":" + password;
             String hashA1 = toHexString(digester.digest(A1.getBytes(StandardCharsets.ISO_8859_1)));
 
-            URI uri = request.getURI();
-            String A2 = request.getMethod() + ":" + uri;
+            String uri = (request.getQuery() == null) ? request.getPath() : String.format("%s?%s", request.getPath(), request.getQuery());
```

Comment:
Factor out `request.getQuery()` into a local variable. Same for `request.getPath()`. No parens in the first expression of the conditional operator.
Keep string concatenation rather than using `String.format()` as it's only 3 strings to concatenate, it's clearer to read and better optimized by the JVM.